### ISSUE TITLE
Forbid imports / calls, per file

### ISSUE THREAD

**Issue by bric3** [detekt/detekt#7827]
## Expected Behavior

I have one file that is should not use some types to avoid `NoClassDefFoundError`. THe import are allowed in other classes.

Could it make sense to have a file annotation that tells imports are forbidden (or even allowed in this file only)

E.g. in pseudo code

```kotlin
@file:ForbiddenImport(
    forbidden = ["git4idea.*"]
)
```

## Current Behavior

Not supported today

**Comment by cortinico** [comment]
> I have one file that is should not use some types to avoid `NoClassDefFoundError`. THe import are allowed in other classes.

Sorry I'm not sure I understand what you're saying here. Could you elaborate further?

There is already a rule called `ForbiddenImport` which does exactly that.

**Comment by bric3** [comment]
I'm aware of `ForbiddenImport` and the others, however this rule is global

``
ForbiddenImport:
  active: true
  imports: []
  forbiddenPatterns: '^it.unimi.dsi.fastutil.*'
```

I would like to apply such a rule in a single class / file only.

**Comment by cortinico** [comment]
This will be possible in detekt 2.0 where you can instantiate a rule multiple times.

**Comment by BraisGabin** [comment]
Closing for the reason mentioned by Nico
